### PR TITLE
Add docs for delta parameters extracted from HiveConfig

### PR DESCRIPTION
**Pull request by homar**
## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 1]
```diff
@@ -126,6 +126,17 @@ values. Typical usage does not require you to configure them.
       - Enables :ref:`Table statistics <delta-lake-table-statistics>` for
         performance improvements.
       - ``true``
+    * - ``delta.per-transaction-metastore-cache-maximum-size``
+      - Maximum number of metastore data objects per transaction in
+        the Hive metastore cache
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/delta-lake.rst:131]: nit: Append `.` for consistency with other properties.

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 2]
```diff
@@ -126,6 +126,17 @@ values. Typical usage does not require you to configure them.
...
+      - ``1000``
+    * - ``delta.delete-schema-locations-fallback``
+      - Whether schema locations should be deleted when Trino can't
+        determine whether they contain external files.
+      - ``false``
+    * - ``delta.parquet.time-zone``
+      - Time zone for Parquet read and write
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/delta-lake.rst:138]: nit: Append `.` for consistency with other properties.

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 3]
```diff
@@ -126,6 +126,17 @@ values. Typical usage does not require you to configure them.
...
+      - ``system default time zone``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/delta-lake.rst:139]: How about "JVM default" as Hive documentation? 
```suggestion
      - JVM default
```

**Review by ebyhr — APPROVED** [review]
Could you drop first two commits? I think they were accidentally introduced during release process.

File: docs/src/main/sphinx/connector/delta-lake.rst (hunk 3 above)
Comment (homar) [reply docs/src/main/sphinx/connector/delta-lake.rst:139]: great idea, thanks

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**